src/components/Engines/Engines.test.tsx

```diff
@@ -1,71 +1,66 @@
 import React from 'react';
 import { mount } from 'enzyme';
 
+import { DetailContext } from '../../pages/Version';
+import { PackageMetaInterface } from '../../../types/packageMeta';
+
 import Engine from './Engines';
 
 jest.mock('./img/node.png', () => '');
 jest.mock('../Install/img/npm.svg', () => '');
 
-const mockPackageMeta: jest.Mock = jest.fn(() => ({
+const mockPackageMeta = (): PackageMetaInterface => ({
   latest: {
-    homepage: 'https://verdaccio.tld',
-    bugs: {
-      url: 'https://verdaccio.tld/bugs',
-    },
+    name: 'verdaccio',
+    version: '0.0.0',
     dist: {
-      tarball: 'https://verdaccio.tld/download',
+      fileCount: 1,
+      unpackedSize: 1,
     },
   },
-}));
-
-jest.mock('../../pages/Version', () => ({
-  DetailContextConsumer: component => {
-    return component.children({ packageMeta: mockPackageMeta() });
-  },
-}));
+  _uplinks: {},
+});
 
 describe('<Engines /> component', () => {
-  beforeEach(() => {
-    jest.resetAllMocks();
-  });
-
   test('should render the component in default state', () => {
-    const packageMeta = {
-      latest: {
-        engines: {
-          node: '>= 0.1.98',
-          npm: '>3',
-        },
-      },
-    };
+    const packageMeta = mockPackageMeta();
 
-    mockPackageMeta.mockImplementation(() => packageMeta);
+    packageMeta.latest.engines = {
+      node: '>= 0.1.98',
+      npm: '>3',
+    };
 
-    const wrapper = mount(<Engine />);
+    const wrapper = mount(
+      <DetailContext.Provider value={{ packageMeta }}>
+        <Engine />
+      </DetailContext.Provider>
+    );
     expect(wrapper.html()).toMatchSnapshot();
   });
 
   test('should render the component when there is no engine key in package meta', () => {
-    const packageMeta = {
-      latest: {},
-    };
+    const packageMeta = mockPackageMeta();
 
-    mockPackageMeta.mockImplementation(() => packageMeta);
+    delete packageMeta.latest.engines;
```

Comment:
Hmm...here you are mutating the object.. something I wouldn't do... I would filter it

Reply:
@bighuggies you can try with `Object.assign` or spread the props, so you don't have to mutate the object.

```
const packageMeta = Object.assign({}, mockPackageMeta(), {
  latest: { engines: undefined })
};
```
This is more verbose, but, according the standard we promote here, no mutations.

Reply:
Because it's a deeply nested object, the non-mutational form would have to be a bit more complicated:

```typescript
    const initialPackageMeta = mockPackageMeta();
    const packageMeta = {
      ...initialPackageMeta,
      ...{
        latest: {
          ...initialPackageMeta.latest,
          engines: undefined,
        },
      },
    };
```

I think the mutational form is easier to read and conveys the intention of the test much more clearly.

Mutations are bad practice in general but I think tests like this are an exception. They run in isolation and the point is to be readable.

Regardless, I've changed `mockPackageMeta` to take the `engine` object as a param, I think it's slightly less clear but avoids mutations.